week-1/artists#logical/tic_tac_toe_improved.rb

```diff
@@ -0,0 +1,164 @@
+require 'colorize'
+#Variables generales
+step_jugadas = 0
+array_no_opciones = []
+
+def creation_solution
+    index_final = @num_board**2 - 1
+    @array_jugadas_winner = []
+    
+    arr_vertical = Array.new(@num_board) { Array.new }
+    con_v = 0
+
+    arr_horizontal = []
+
+    arr_diagonal = Array.new(2) { Array.new }
+    
+    for i in 0..index_final
+        #Creando soluciones del michi horizontales
+        arr_horizontal.push(i)
+
+        #Creando soluciones del michi verticales
+        arr_vertical[con_v].push(i)
+        con_v += 1
+
+        #Creando soluciones del michi diagonales
+        arr_diagonal[0].push(i) if i % (@num_board + 1) == 0
+        arr_diagonal[1].push(i) if i > 0 && i < index_final && (i % (@num_board - 1) == 0)
+
+        if (i+1) % @num_board == 0
+            @array_jugadas_winner.push(arr_horizontal)
+            arr_horizontal=[]
+            con_v = 0
+        end
+    end
+    # puts @array_jugadas_winner.to_s
+    # puts arr_vertical.to_s
+    # puts arr_diagonal.to_s
+
+    @array_jugadas_winner = @array_jugadas_winner + arr_vertical + arr_diagonal
+end
+
+def creation_board (arr)
+    num_arr=arr.length - 1
+    tablero = ""
+    for i in (0..num_arr)
+        posicion = "  " + (i+1).to_s
+        move = false
+        case arr[i]
+            when "1"
+                posicion = "  x".blue
+                move = true
+            when "0"
+                posicion = "  o".red
+                move = true
+        end
+        
+        if i < num_arr
+            if (i + 1) % @num_board == 0
+                posicion += "\n" + "-"*(@num_board*6-1) +"\n"
+            else
+                posicion += " "
+                if !move
+                    if (i+1)>9
+                        posicion += "|"
+                    else
+                        posicion += " |"
+                    end
```

Comment:
I would suggest you to change those lines of code to one line using a ternary conditional, because it would be easier to read for beginners.

Reply:
Thanks! I am going to consider next time, it could be very helpful.